### PR TITLE
Added timeout feature for AutoInstall

### PR DESCRIPTION
**Pull request by toledo-labs**
## Types of changes
I added an update timeout method whenever AutoInstall is used. The default timeout is updated from 30 seconds to 1 second in the grub file. 

## Expected behavior
If AutoInstall in enabled the boot option menu appears for 1 second instead of 30. If this period is too short I'll change it in the code review.

### CODE REVIEW

File: usn/ubuntu_server_netboot.py
Comment (dannf) [inline usn/ubuntu_server_netboot.py]: I'm concerned about the future proofing of the `timeout=` matching here. For one, a future installer may include a default different than `30`. For another this will match other `*timeout=30` locations, say if a future installer shipped w/ `deferred_probe_timeout=30` by default. Would it be possible to rework this to match a regex of something like `r'^set timeout=\d+'`, perhaps using something like `re.sub`?

**Comment by dannf** [comment]
Thanks - this does look valuable. One comment inline - also, would you mind updating the tests? I think this would make the existing one fail since the expected result still has `set timeout=30`

**Comment by toledo-labs** [comment]
Yes, I'll update the test and clear the line before setting the timeout=1. This way the solution will work for any timeout if set other than 30 in the future.

**Comment by toledo-labs** [comment]
The app was adding trailing newline characters in the grub file in both methods: add_kernel_params and update_timeout. Original ISO grubs files don't have trailing newline characters so I removed them in both routines. 

Screenshot from 2023-10-16 17-16-06

**Review by dannf — CHANGES_REQUESTED** [review]
Oops, apparently I didn't add my inline comment. Hopefully this does it.
Thank you for addressing the extra whitespace issue! However, would you mind doing that in a separate commit? Ideally one commit to fix the original issue w/o the timeout feature (both code and test), and then an additional commit that adds the feature and the necessary test tweak together. If that's complicated, I'm happy to munge the commits myself before merging.

File: usn/ubuntu_server_netboot.py
Comment (toledo-labs) [reply usn/ubuntu_server_netboot.py]: Yes I agree with the regex solution.

**Comment by dannf** [comment]
Thanks - this is looking very close to ready. Would you mind removing the whitespace changes (mostly `\n` changes) to `usn/ubuntu_server_netboot.py` in commit 88b69328a4ce7cc6bc8f35fab95db75312d67663 ?

**Comment by toledo-labs** [comment]
I think I removed all the whitespace changes and amend the last commit.

File: usn/ubuntu_server_netboot.py
Comment (dannf) [inline usn/ubuntu_server_netboot.py]: This is one of the whitespace changes I'm referring to

File: usn/ubuntu_server_netboot.py
Comment (dannf) [inline usn/ubuntu_server_netboot.py]: And this one

File: usn/ubuntu_server_netboot.py
Comment (dannf) [inline usn/ubuntu_server_netboot.py]: And this one

**Review by dannf — CHANGES_REQUESTED** [review]
Sorry, I meant the whitespace in the first commit - I've added inline comments to be clear.

File: usn/ubuntu_server_netboot.py
Comment (toledo-labs) [reply usn/ubuntu_server_netboot.py]: Removed on the last commit.

File: usn/ubuntu_server_netboot.py
Comment (toledo-labs) [reply usn/ubuntu_server_netboot.py]: This change isn't present on the final solution. The timeout feature required the fixing of the whitespace in the expected config along with the  two routines. The two issues are related and therefore in my opinion should be in the same commit. I can squash the two commits into one for the sake of time. Otherwise I leave it to you to make the final changes. I hope I've added value to your tool. Thank you.

Comment (dannf) [reply usn/ubuntu_server_netboot.py]: No problem - I think we might just be miscommunicating. I've gone ahead and made the changes I was trying to highlight and merged (removing arbitrary whitespace changes in the code itself, unrelated to the whitespace fix you contributed). I also noticed an optimization with re.sub we could make, so I've added that as a subsequent commit.

Thank you for your patience, and thank you for this feature @toledo-systems !

**Pull request closed without merging**
